util/gcp-tools/src/main/kotlin/ai/tock/gcp/secretmanager/GcpSecretManagerService.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2017/2021 e-voyageurs technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.gcp.secretmanager
+
+import ai.tock.gcp.EnvConfig
+import ai.tock.gcp.GCP_SECRET_VERSION
+import ai.tock.shared.security.SecretManagerProviderType
+import ai.tock.shared.security.SecretManagerService
+import ai.tock.shared.security.credentials.AIProviderSecret
+import ai.tock.shared.security.credentials.Credentials
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.google.api.gax.rpc.AlreadyExistsException
+import com.google.cloud.secretmanager.v1.*
+import com.google.protobuf.ByteString
+import kotlinx.serialization.json.Json
+import mu.KLogger
+import mu.KotlinLogging
+
+/**
+ * Implementation of the AWS Secret Manager Service
+ */
+class GcpSecretManagerService: SecretManagerService {
+
+    override val type: SecretManagerProviderType
+        get() = SecretManagerProviderType.GCP_SECRET_MANAGER
+
+    private val client: SecretManagerServiceClient = SecretManagerServiceClient.create()
+    private val logger: KLogger = KotlinLogging.logger { }
+
+    // Get an existing secret.
+    private fun getGcpSecret(secretId: String): String  {
+        // Access the secret version.
+        val secretVersionName: SecretVersionName = SecretVersionName.of(EnvConfig.gcpProjectId, secretId, GCP_SECRET_VERSION)
```

Comment:
I don't think it makes sense, because secrets evolve independently. In my opinion, it should also be removed on the AWS side. I don't see any use case where we're going to force all secrets to a given version, whereas via Tock Studio we've given the possibility to modify the value of the secret and thus evolve its version. what do you think?

Reply:
Agreed, I don't have any use case in mind were it would be useful